File: examples/core_input_keys.c
/*******************************************************************************************
*
*   raylib example 03a - Keyboard input 
*
*   This example has been created using raylib 1.0 (www.raylib.com)
*   raylib is licensed under an unmodified zlib/libpng license (View raylib.h for details)
*
*   Copyright (c) 2013 Ramon Santamaria (Ray San - raysan@raysanweb.com)
*
********************************************************************************************/

#include "raylib.h"

int main()
{    
    // Initialization
    //--------------------------------------------------------------------------------------
    int screenWidth = 800;
    int screenHeight = 450;
    
    Vector2 ballPosition = { screenWidth/2, screenHeight/2 };
    
    InitWindow(screenWidth, screenHeight, "raylib example 05 - keyboard input");
    
    SetTargetFPS(60);       // Set target frames-per-second
    //--------------------------------------------------------------------------------------
    
    // Main game loop
    while (!WindowShouldClose())    // Detect window close button or ESC key
    {
        // Update
        //----------------------------------------------------------------------------------
        if (IsKeyDown(KEY_RIGHT)) ballPosition.x += 0.8;
        if (IsKeyDown(KEY_LEFT)) ballPosition.x -= 0.8;
        if (IsKeyDown(KEY_UP)) ballPosition.y -= 0.8;
        if (IsKeyDown(KEY_DOWN)) ballPosition.y += 0.8;
        //----------------------------------------------------------------------------------
        
        // Draw
        //----------------------------------------------------------------------------------
        BeginDrawing();
        
            ClearBackground(RAYWHITE);
            
            DrawText("move the ball with arrow keys", 10, 10, 20, DARKGRAY);
            
            DrawCircleV(ballPosition, 50, MAROON);
        
        EndDrawing();
        //----------------------------------------------------------------------------------
    }

    // De-Initialization
    //--------------------------------------------------------------------------------------
    CloseWindow();        // Close window and OpenGL context
    //--------------------------------------------------------------------------------------
    
    return 0;
}

File: examples/core_input_keys.lua
-------------------------------------------------------------------------------------------
--
--  raylib [core] example - Keyboard input
--
--  This example has been created using raylib 1.6 (www.raylib.com)
--  raylib is licensed under an unmodified zlib/libpng license (View raylib.h for details)
--
--  Copyright (c) 2014-2016 Ramon Santamaria (@raysan5)
--
-------------------------------------------------------------------------------------------

-- Initialization
-------------------------------------------------------------------------------------------
local screenWidth = 800
local screenHeight = 450

InitWindow(screenWidth, screenHeight, "raylib [core] example - basic window")

local ballPosition = Vector2(screenWidth/2, screenHeight/2)

SetTargetFPS(60)                -- Set target frames-per-second
-------------------------------------------------------------------------------------------

-- Main game loop
while not WindowShouldClose() do                -- Detect window close button or ESC key
    -- Update
    ---------------------------------------------------------------------------------------
    if (IsKeyDown(KEY.RIGHT)) then ballPosition.x = ballPosition.x + 0.8 end
    if (IsKeyDown(KEY.LEFT)) then ballPosition.x = ballPosition.x - 0.8 end
    if (IsKeyDown(KEY.UP)) then ballPosition.y = ballPosition.y - 0.8 end
    if (IsKeyDown(KEY.DOWN)) then ballPosition.y = ballPosition.y + 0.8 end
    ---------------------------------------------------------------------------------------

    -- Draw
    ---------------------------------------------------------------------------------------
    BeginDrawing()

        ClearBackground(RAYWHITE)

        DrawText("move the ball with arrow keys", 10, 10, 20, DARKGRAY)

        DrawCircleV(ballPosition, 50, MAROON)

    EndDrawing()
    ---------------------------------------------------------------------------------------
end

-- De-Initialization
-------------------------------------------------------------------------------------------
CloseWindow()           -- Close window and OpenGL context
-------------------------------------------------------------------------------------------
